portfolio/src/main/webapp/script.js

```diff
@@ -29,73 +29,138 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
+/**
+ * Composes a higher order function with world
+ * @param {function} f - a function with @param {string} which @returns {function}
+ * @return {function} a composition with 'world' and the result of f
+ */
 function composeWorld(f) {
   return f('world');
 }
 
+/**
+ * Returns a greeting function to a person
+ * @param {string} name - the person to be greeted
+ * @return {function} a function which has @param {string} punctuation
+ *                    which then greets a person with {name} upon being called 
+ */
 function greet(name) {
   return (punctuation) => `Hello ${name}${punctuation}`;
 }
 
+/**
+ * Returns a happier version of an inputted string
+ * @param {string} item - a body of text
+ * @return {string} @param item with the addition of a laughing emoji
+ */
 function addHappiness(item) {
   return item + ' 😂';
 }
 
+/**
+ * Updates the max number of comments which are showing
+ * @return {void}
+ */
 async function updateMaxComments() {
   const maxComments = document.querySelector("#maxComments--number-input").value;
-  defaultMaxComments = maxComments; 
+  defaultMaxComments = maxComments;
   await getData(maxComments);
 }
 
+/**
+ * Deletes all comments which are stored in the datastore
+ */
 async function deleteComments() {
-    try {
-        const ret = await fetch('/delete-data', {
-            method: "POST"
-        })
-        alert("A truly sad day, your comments were deleted");
-        await getData();
-    } catch (e) {
-        alert("Something went wrong in deleting your comments");
-    }
+  try {
+    const ret = await fetch('/delete-data', {
+      method: "POST"
+    })
+    alert("A truly sad day, your comments were deleted");
+    await getData();
+  } catch (e) {
+    alert("Something went wrong in deleting your comments");
+  }
+}
+
+/**
+ * Get the comments from the server and populate the comment DIV
+ * @param {number?} maxComments - the max number of comments to display
+ */
+async function getData(maxComments) {
```

Comment:
I believe you can set the default for optional variables here by doing:
async function getData(maxComments = defaultMaxComments) {
...
}

This should only set it to defaultMaxComments if maxComments isn't explicitly passed into the function